src/in_process_server.cpp

```diff
@@ -806,6 +796,14 @@ class wlcs::Client::Impl
         return xdg_shell_stable;
     }
 
+    zwlr_layer_shell_v1* the_layer_shell_v1() const
+    {
+        if (layer_shell_v1)
+            return layer_shell_v1;
+        else
+            BOOST_THROW_EXCEPTION((ExtensionExpectedlyNotSupported{"zwlr_layer_shell_v1", 0}));
```

Comment:
This is wrong.

`ExtensionExpectedlyNotSupported` is used elsewhere to indicate that the extension is not in the `supported_extensions`provided by the test fixture.

Here we've discovered that the server doesn't publish `layer_shell_v1` as a global, but we are not checking `supported_extensions`. IMO the test for `supported_extensions` should be earlier: if we get to here then we should be failing.

See, for example, the use of `CheckInterfaceExpected` in recent test fixtures.